janitor/functions.py

```diff
@@ -4590,3 +4594,284 @@ def complete(
+    `.value` as new column names.
+
+    .. code-block:: python
+
+        df = (pd.DataFrame(...)
+              .pivot_longer(names_to=("group",'.value'),
+                            names_sep = '-')
+              )
+
+            group  measure1  measure2
+        0  treat1         1         4
+        1  treat2         2         5
+        2  treat1         2         5
+        3  treat2         3         4
+
+    Example 4: We can also pivot from wide to long using regular expressions
+
+    .. code-block:: python
+
+            n_1  n_2  n_3  pct_1  pct_2  pct_3
+        0   10   20   30   0.1    0.2    0.3
+
+        df = (pd.DataFrame(...)
+              .pivot_longer(names_to = (".value", "name"),
+                            names_pattern = "(.*)_(.)")
+              )
+
+            name    n  pct
+        0     1  10.0  0.1
+        1     2  20.0  0.2
+        2     3  30.0  0.3
+
+    You can also take advantage of `janitor.patterns` function, which allows
+    selection of columns via a regular expression; this can come in handy if
+    you have a lot of column names to use as index, and do not wish to manually
+    type them all.
+
+    .. code-block:: python
+
+             name    wk1   wk2   wk3   wk4
+        0    Alice     5     9    20    22
+        1    Bob       7    11    17    33
+        2    Carla     6    13    39    40
+
+        df = pd.DataFrame(...).pivot_longer(janitor.patterns("^(?!wk)"))
+
+             name variable  value
+        0   Alice      wk1      5
+        1   Alice      wk2      9
+        2   Alice      wk3     20
+        3   Alice      wk4     22
+        4     Bob      wk1      7
+        5     Bob      wk2     11
+        6     Bob      wk3     17
+        7     Bob      wk4     33
+        8   Carla      wk1      6
+        9   Carla      wk2     13
+        10  Carla      wk3     39
+        11  Carla      wk4     40
+
+    Functional usage syntax:
+
+    .. code-block:: python
+
+        import pandas as pd
+        import janitor as jn
+
+        df = pd.DataFrame(...)
+        df = jn.pivot_longer(
+            df = df,
+            index = [column1, column2, ...],
+            column_names = [column3, column4, ...],
+            names_to = new_column_name,
+            names_sep = string or regular expression,
+            names_pattern =s tring or regular expression,
+            value_name = new_column_name
+        )
+
+    Method chaining syntax:
+
+    .. code-block:: python
+
+        df = (
+            pd.DataFrame(...)
+            .pivot_longer(df,
+                         index = [column1, column2, ...],
+                         column_names = [column3, column4, ...],
+                         names_to = new_column_name,
+                         names_sep=string or regular expression,
+                         names_pattern=string or regular expression,
+                         value_name= new_column_name)
```

Comment:
```suggestion
            .pivot_longer(
                df,
                index = [column1, column2, ...],
                column_names = [column3, column4, ...],
                names_to = new_column_name,
                names_sep=string or regular expression,
                names_pattern=string or regular expression,
                value_name= new_column_name
            )
        )
```

Reply:
how do you get the format working again? on vscode? I'll save the guide this time so you dont tell me for the 200th time.